src/ast/compare.rs

```diff
@@ -59,10 +59,6 @@ pub enum Compare<'a> {
 pub enum JsonCompare<'a> {
     ArrayContains(Box<Expression<'a>>, Box<Expression<'a>>),
     ArrayNotContains(Box<Expression<'a>>, Box<Expression<'a>>),
-    ArrayBeginsWith(Box<Expression<'a>>, Box<Expression<'a>>),
-    ArrayNotBeginsWith(Box<Expression<'a>>, Box<Expression<'a>>),
-    ArrayEndsInto(Box<Expression<'a>>, Box<Expression<'a>>),
-    ArrayNotEndsInto(Box<Expression<'a>>, Box<Expression<'a>>),
```

Comment:
As explained above, this means we no longer need these `JsonCompare` variants as we now rely on `Compare::Equals` & `Compare::NotEquals`.